legacy/firmware/signing.c

```diff
@@ -519,19 +519,14 @@ void signing_init(const SignTx *msg, const CoinInfo *_coin,
   expiry = (coin->decred || coin->overwintered) ? msg->expiry : 0;
   timestamp = coin->timestamp ? msg->timestamp : 0;
   if (coin->overwintered) {
+    if (version != 4 || !msg->has_branch_id) {
```

Comment:
How about `version_group_id`? We do check that for Core

Reply:
We validate it [couple of lines above](https://github.com/trezor/trezor-firmware/blob/ed31ad4418ff37975213c959b7ec34872e6f44fb/legacy/firmware/signing.c#L497).  I suppose it would make sense to have all zcash-related check in one place (and inside `#if !BITCOIN_ONLY`).

Reply:
That'd be nice!

Reply:
Fixed 0caf0165bf2cec2bd2e90a025572b19f324c4fca.